components/indexer.py

```diff
@@ -1,56 +1,91 @@
 import rev
 import wpilib
+from enum import Enum
 from magicbot import tunable, feedback
-from typing import Tuple
 
 
 class Indexer:
-    indexer_front_motor: rev.CANSparkMax
-    indexer_back_motor: rev.CANSparkMax
+    class Direction(Enum):
+        OFF = 0
+        FORWARDS = 1
+        BACKWARDS = -1
+
+    # The "tunnel" is the horizontal part of the indexer that the cargo enters first
+    # The "chimney" is the vertical section of the indexer that feeds the shooter
+    # The "cat flap" is the moving flap that allows an opposition cargo to be captured
+    indexer_tunnel_motor: rev.CANSparkMax
+    indexer_chimney_motor: rev.CANSparkMax
     colour_sensor: rev.ColorSensorV3
-    prox_sensor1: wpilib.DigitalInput
-    prox_sensor2: wpilib.DigitalInput
-    # indexer_piston: wpilib.Solenoid
+    lower_chimney_prox_sensor: wpilib.DigitalInput
+    upper_chimney_prox_sensor: wpilib.DigitalInput
+    cat_flap_piston: wpilib.DoubleSolenoid
 
     is_firing = tunable(False)
     is_red = tunable(False)
     indexer_speed = tunable(0.5)
-    # Front, back
-    speeds = (0.0, 0.0)
+
+    _tunnel_direction = Direction.OFF
+    _chimney_direction = Direction.OFF
+    _cat_flap_is_open = False
+
+    has_trapped_cargo = tunable(False)
 
     def setup(self) -> None:
-        self.indexer_front_motor.setInverted(True)
-        self.indexer_back_motor.setInverted(False)
+        self.indexer_tunnel_motor.setInverted(True)
+        self.indexer_chimney_motor.setInverted(False)
 
     def execute(self) -> None:
-        self.indexer_front_motor.set(self.speeds[0])
-        self.indexer_back_motor.set(self.speeds[1])
+        self.indexer_tunnel_motor.set(self.indexer_speed * self._tunnel_direction)
+        self.indexer_chimney_motor.set(self.indexer_speed * self._chimney_direction)
```

Comment:
mypy errors on these lines:
```
components/indexer.py:38: error: Unsupported operand types for * ("float" and "Direction")
components/indexer.py:39: error: Unsupported operand types for * ("float" and "Direction")
```
Have you tried running this in the simulator? :)

```suggestion
        self.indexer_tunnel_motor.set(self.indexer_speed * self._tunnel_direction.value)
        self.indexer_chimney_motor.set(self.indexer_speed * self._chimney_direction.value)
```